lib/psci/psci_private.h

```diff
@@ -38,6 +38,29 @@
 #include <psci.h>
 #include <spinlock.h>
 
+#if HW_ASSISTED_COHERENCY
+/*
+ * On systems with hardware-assisted coherency, make PSCI cache operations NOP,
+ * as PSCI participants are cache-coherent, and there's no need for explicit
+ * cache maintenance operations or barriers to coordinate their state.
+ */
+#define psci_flush_dcache_range(addr, size)
```

Comment:
Sorry, this comment got masked in the PR. IMO, this macro is a simple wrapper to the underlying function with both the parameters being passed directly to the underlying function. Hence, in case HW_ASSISTED_COHERENCY is defined, the function prototype of flush_dache_range ensures the type checking of the parameters to this macro. When HW_ASSISTED_COHERENCY is not defined, it is a NOP and it doesnt matter anyway. Hence I think the macro achieves the types enforcement when it matters.